tests/compiletest.rs

```diff
@@ -145,7 +145,10 @@ fn run_tests(mode: Mode, path: &str, target: &str, with_dependencies: bool) -> R
         // The files we're actually interested in (all `.rs` files).
         |path| {
             path.extension().is_some_and(|ext| ext == "rs")
-                && (filters.is_empty() || filters.iter().any(|f| path.starts_with(f)))
+                && (filters.is_empty()
+                    || filters.iter().any(|f| {
+                        path.components().any(|c| c.as_os_str().to_string_lossy().contains(f))
```

Comment:
That's not the right behavior -- filtering for `pass/stack` should also work. It should just be doing substring on the entire filename.